packages/payment-processor/src/payment/utils-near.ts

```diff
@@ -28,14 +29,29 @@ export const isValidNearAddress = async (nearNetwork: Near, address: string): Pr
 export const isNearAccountSolvent = (
   amount: BigNumberish,
   nearWalletConnection: WalletConnection,
+  token?: RequestLogicTypes.ICurrency,
 ): Promise<boolean> => {
-  return nearWalletConnection
-    .account()
-    .state()
-    .then((state) => {
-      const balance = BigNumber.from(state?.amount ?? '0');
-      return balance.gte(amount);
-    });
+  if (!token || token.type === RequestLogicTypes.CURRENCY.ETH) {
+    return nearWalletConnection
+      .account()
+      .state()
+      .then((state) => {
+        const balance = BigNumber.from(state?.amount ?? '0');
+        return balance.gte(amount);
+      });
+  }
+  if (token.type === RequestLogicTypes.CURRENCY.ERC20) {
```

Comment:
In this case, shouldn't we also check the account solvency in native currency for:
- The transaction costs
- The mandatory deposit of 1 yoctoNEAR

If you add the condition, having the associated test would be nice

Reply:
That's a very good point, I did not think of that. The wallet UX is quite good when it comes to lack of NEAR (for both fees and deposit), so I think it's OK to let it handle this case.

Reply:
@leoslr  cc @alexandre-abrioux